src/coreclr/inc/corinfo.h

```diff
@@ -2265,6 +2265,27 @@ class ICorStaticInfo
             int                         bufferSize  /* IN  */
             ) = 0;
 
+
+    //------------------------------------------------------------------------------
+    // printObjectDescription: Prints a (possibly truncated) textual UTF8 representation of the given
+    //    object to a preallocated buffer. It's intended to use only for debug/diagnostic 
+    //    purposes such as JitDisasm. The buffer is null-terminated (even if truncated).
+    //
+    // Arguments:
+    //    handle     - Direct object handle
+    //    buffer     - Pointer to buffer
+    //    bufferSize - Pointer to buffer length. Must not be nullptr
+    //
+    // Return Value:
+    //    Full length of the source UTF8 representation so can be bigger than bufferSize.
+    //    Includes null-terminator.
```

Comment:
```suggestion
    //    Full length of the source UTF8 representation so can be bigger than bufferSize.
    //    Includes null-terminator.
```
This is somewhat unusual convention for an API that takes buffer and returns `size_t` (e.g. sprintf_s). Is there a standard C API that has the same convention?

Reply:
`getStringLiteral` API is not pretty printing API. I do not see a problem with it. (BTW: We may want to change the `getStringLiteral` signature to take offset argument, so that we do not need to copy up to 1024 characters and immediately throw away most of it except the one character. Separate PR.)

I was commenting on the fact that it returns the length of the string plus the null terminator. I believe that the usual convention is to return the number of characters written into the buffer without the null terminator. `sprintf` is example of an API that follows this convention.

I do not have a preference between `size_t` and `int`. `size_t` is fine with me.

Reply:
> so that we do not need to copy up to 1024 characters and immediately throw away most of it except the one character. 

That place definitely needs smaller buffer (or rather no buffer at all) no idea how it ended up with 1024, probably was used in JITDUMP initially.
It's also used on other paths where it consumes more characters e.g. StartsWith/Equals unrolling.

>  I believe that the usual convention is to return the number of characters written into the buffer without the null terminator

I agree, can change that in a moment, although, the advantage of the current API is that it easily tells you that you might want to try again with a larger buffer if you want to print the whole thing (so you can heap-allocate next chunk to the exact size) while with `bytesWritten` it's not clear -- was the string exactly 64 bytes? how many bytes should I try next attempt? 

But it all doesn't matter for this particular API since it's only supposed to be used in debug purposes and doesn't need to print the whole thing if it's too long.

Reply:
Excluding null terminator is also fine with me, it seems better if that is more in line with the sprintf_s family of APIs. I still think we should return the number of bytes in the actual string (i.e., calling again with a buffer of sized `returnedSize + 1` will get the full string).

Reply:
I assume in general there is no a standardized shape, e.g. APIs which work on streamed data can not know the final length in advance (a good example is that `Utf8.FromUtf16` that returns status of a operation e.g. `DestinationTooSmall`)